dotcom-rendering/src/web/client/performanceMonitoring.ts

```diff
@@ -0,0 +1,94 @@
+import { log } from '@guardian/libs';
+import { record } from './ophan/ophan';
+
+const logPerformanceInfo = (name: string, data?: unknown) =>
+	log('openJournalism', '⏱', name, data);
+
+const reasons = new Set<string>();
+const recordPoorDeviceConnectivity = (reason: string) => {
+	reasons.add(reason);
+	log('openJournalism', '🐌', reasons);
+	if (reasons.size >= 2) {
+		/** Not sure here if we should duplicate “dotcom-rendering” */
+		const experiences = ['dotcom-rendering', 'poor-page-performance'].join(
+			',',
+		);
+		record({ experiences });
+	}
+};
+
+export const performanceMonitoring = (): Promise<void> => {
+	try {
+		// First contentful paint is the best single indicator
+		// of core web vital label
+		new PerformanceObserver((entries, observer) => {
+			for (const { name, duration, startTime } of entries.getEntries()) {
+				switch (name) {
+					case 'first-contentful-paint': {
+						logPerformanceInfo(name, { startTime, duration });
+						if (startTime > 2400) {
+							recordPoorDeviceConnectivity('FCP over 2.4s');
+							observer.disconnect();
+						}
+					}
+				}
+			}
+		}).observe({
+			type: 'paint',
+			buffered: true,
+		});
+
+		// Time to first byte is a widely supported performance marker
+		// but its value is
```

Comment:
I have a feeling there's part of this comment missing, but maybe I'm reading it incorrectly!